framework/base/Security.php

```diff
@@ -305,7 +305,7 @@ public function hkdf($algo, $inputKey, $salt = null, $info = null, $length = 0)
         if (is_string($length) && preg_match('{^\d{1,16}$}', $length)) {
             $length = (int) $length;
         }
-        if (!is_integer($length) || $length < 0 || $length > 255 * $hashLength) {
+        if ($length < 0 || $length > 255 * $hashLength || !is_integer($length)) {
```

Comment:
How ever rearrange can affect readability? I see no difference in readability at all.